app/services/setting_registry.rb

```diff
@@ -77,6 +80,8 @@ def load_values
     end
     # load nil to set value to default
     @settings.except(*loaded_names).each do |name, definition|
+      # Creating missing records as we operate over the DB model while updating the setting
+      _new_db_record(definition).save(validate: false)
       definition.updated_at = nil
```

Comment:
Are these changes (the three above) necessary for the DSL? If this PR would only add the new DSL we could easily introduce it even into 2.5 and present as the new way of defining settings.

Reply:
It would not be necessary if we merge #8438, otherwise the first setting added to the DSL would be missing from DB and thus from the UI.
So we would need to cherry-pick the change for UI rendering, which would make me happy to have as that is quite huge blocker for this whole effort, it seems to be quite big change at this time.